clang/test/SemaTemplate/deduction.cpp

```diff
@@ -431,6 +442,17 @@ namespace deduction_after_explicit_pack {
     i<int, int>(0, 1, 2, 3, 4, 5); // expected-error {{no match}}
   }
 
+  template <typename... T>
+  void bar(args_tag<T...>, type_identity_t<T>..., int mid, type_identity_t<T>...) {}
```

Comment:
What happens without the mid parameter ?

Reply:
It could work. It would cost only condition being removed. I deliberately put in the condition that disables this. Should I enable it?

I think enabling it would be standard compliant, but I played it safe and stuck to the example seen in the issue.

Reply:
Yes, i think we should support that, and add a test for it )as long as the standard says it should work, which i think it does

Reply:
I added a test for this.

Reply:
I'd like to see a test for ```cpp
template <typename... Y, typename... T>
void foo2(args_tag<Y...>, args_tag<T...>, type_identity_t<T>..., type_identity_t<T>...) {}
```

Reply:
Added the test. It also works.